src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs

```diff
@@ -96,6 +93,81 @@ public async Task TestHasSuccessfullyLoadedBeingFalseWithCompilerAnalyzerFSAOn()
             await TestAnalyzerAsync(workspace, document, new CSharpCompilerDiagnosticAnalyzer(), CompilerAnalyzerResultSetter, expectedSyntax: true, expectedSemantic: false);
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task TestDisabledByDefaultAnalyzerEnabledWithEditorConfig(bool enabledWithEditorconfig)
+        {
+            using var workspace = new AdhocWorkspace();
+            workspace.Options = workspace.Options.WithChangedOption(SolutionCrawlerOptions.BackgroundAnalysisScopeOption, LanguageNames.CSharp, BackgroundAnalysisScope.FullSolution);
+
+            var project = workspace.AddProject(
+                ProjectInfo.Create(
+                    ProjectId.CreateNewId(),
+                    VersionStamp.Create(),
+                    "CSharpProject",
+                    "CSharpProject",
+                    LanguageNames.CSharp,
+                    filePath: "z:\\CSharpProject.csproj"));
+
+            if (enabledWithEditorconfig)
+            {
+                var editorconfigText = @$"
+[*.cs]
+dotnet_diagnostic.{DisabledByDefaultAnalyzer.s_syntaxRule.Id}.severity = warning
+dotnet_diagnostic.{DisabledByDefaultAnalyzer.s_semanticRule.Id}.severity = warning
+dotnet_diagnostic.{DisabledByDefaultAnalyzer.s_compilationRule.Id}.severity = warning";
```

Comment:
💡 Consider testing each of these IDs separately

💡 Consider testing with other severities

💡 If you have several parameters, you can use `[PairwiseData]` instead of `[CombinatorialData]` to test many combinations without state explosion

Reply:
The intent here is just to test that disabled analyzers can be enabled from editorconfig, so I am not sure if these additional combinations add value for this test. It validates that each of the diagnostic incremental analyzer callbacks (AnalyzeSyntaxAsync, AnalyzeDocumentAsync and AnalyzerProjectAsync) have the correct behavior, hence the three rules. I have also kept testing for all IDs in the same test to match existing tests in this file.